packages/back-end/src/validators/metric-time-series.ts

```diff
@@ -0,0 +1,88 @@
+import { z } from "zod";
+
+const metricTimeSeriesValue = z
+  .object({
+    value: z.number(),
+    denominator: z.number().optional(),
+    expected: z.number().optional(),
+    ci: z.tuple([z.number(), z.number()]).optional(),
+    pValue: z.number().optional(),
+    pValueAdjusted: z.number().optional(),
+    chanceToWin: z.number().optional(),
+  })
+  .strict();
+export type MetricTimeSeriesValue = z.infer<typeof metricTimeSeriesValue>;
+
+const metricTimeSeriesVariation = z.object({
+  id: z.string(),
+  name: z.string(),
```

Comment:
How much would this save? If the `id` is not the index then we can rely on it and remove `name`, otherwise we need to figure out something else.